### ISSUE TITLE
Restart on Remote Administration WSC Fixed!

### ISSUE THREAD

**Issue by KSilverman** [matt-cook/kiosk#235]
I've seen a lot of issues revolving around the remote administration not working, and that the updated website does not get updated on Kiosk application that is running on a certain device.

I went into the JScript and fixed it myself. This is just MY version of how to resolve the issue, and I'm sure there are more ways to solve it.

The issue that was happening was that the chrome.runtime.sendMessage() method was not working properly. In order to get it to work, you have to configure it to send an External message to the chrome runtime because we are running a Web Server Chrome (WSC) for our remote administration.

To fix this, first we have to make another message listener for the 'reload' channel because we want to get that channel externally (not from the native client, but rather from a web page). In main.js, add the following code:

![externalmessage](https://cloud.githubusercontent.com/assets/5942751/26827145/e85c7466-4a89-11e7-847d-7d7c4e72823f.PNG)

Again, this is Matt's code, I'm just copying it over to listen on an external channel. Lastly, in the WSC folder directory, navigate to admin.js and add the following code inside the success function for the restart click function right before the window gets reloaded:

![idmessage](https://cloud.githubusercontent.com/assets/5942751/26827216/42548abc-4a8a-11e7-9cf8-d3d721704327.PNG)

And there you go, your Kiosk application will now restart on click! I hope this helps anyone here that is having trouble with the restart functionality, and hopefully a version of this will be implemented on the next release. Thanks! @matt-cook

**Comment by wfstech** [comment]
How/where do I edit the js?

**Comment by KSilverman** [comment]
Clone the github repo, and use a text editor (I use Atom) to go and edit the js. Then, go to your chrome extensions and load the project src as an unpacked extension. That is also where you get the app ID to put in the code.

**Comment by michaelvelasquez** [comment]
Thank you @KSilverman Does this only fix Remote Administration Restart when combined with Web Server Chrome (WSC) for remote administration? 

I have Kiosk Remote Administration working on an ASUS Chromebit on port 8080 on both my local network and from outside my network with port forwarding on my router. I did this just by checking "Remote Administration" within the Kiosk app. I can log in from the browser of another computer, edit and save a URL but I can't restart. 

When I click Restart, the Kiosk Remote Admin page shows a "Restarting" message and then refreshes, but Kiosk doesn't restart on the Chromebit. It doesn't crash, but nothing happens. If I manually reboot Kiosk on the Chromebit (unplug, plug back in), my edited URL displays so it is being saved.

I am not using WSC. I'm not a programmer and I'm not even sure how I'd use WSC. Eventually I want to install Chromebits in kiosk mode running the Kiosk app at remote locations and admin from my office using Remote Administration in the Kiosk app.

I added all your code from the first picture to the end of kiosk-master/src/js/main.js and loaded it onto my Chromebit as an unpacked extension.  It loads and launches without any errors. But I still can't use the Restart button when using Remote Administration.

Thank you.
